R/repr_htmlwidget.r

```diff
@@ -22,3 +22,57 @@ repr_html.htmlwidget <- function(obj, ...) {
 	
 	readChar(htmlfile, file.info(htmlfile)$size)
 }
+
+
+
+# sort of like htmlwidgets::saveWidget(), but for HTML tags
+save_html <- function(obj, ...) {
+	
+	if (!requireNamespace('htmlwidgets')) {
+		stop('Please install the htmlwidgets package')
+	}
+	if (!getFromNamespace("pandoc_available", asNamespace("htmlwidgets"))()) {
+		stop("Printing HTML tags requires pandoc. For details see:\n", 
+				 "https://github.com/rstudio/rmarkdown/blob/master/PANDOC.md")
+	}
+	
+	htmlfile <- tempfile(fileext = '.html')
+	libdir <- file.path(dirname(htmlfile), "lib")
+	htmltools::save_html(obj, file = htmlfile, libdir = libdir)
+	
+	# TODO: it would be nice if we didn't have to rely on pandoc 
+	# https://github.com/rstudio/htmltools/issues/73
+	getFromNamespace("pandoc_self_contained_html", asNamespace("htmlwidgets"))(htmlfile, htmlfile)
+	
+	readChar(htmlfile, file.info(htmlfile)$size)
```

Comment:
Well, as long as we're using `htmlwidgets::pandoc_self_contained_html()`, writing to disk is necessary, and I'm using a tempfile since they automatically get cleaned up when the R session exits...IMO manually cleaning up a tempfile just adds unnecessary complexity